gen_compile_commands

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```

Comment:
What is the point of this file?

Reply:
this file generates a compile_commands.json file for the ccls language server. I use this for syntax checking and autocomplete. It also generates other crap I don't need which is the point of the git clean command. I also will run this command as a lazy way to fix build issues if there are untracked files messing up compilation.